Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -290,6 +289,27 @@ public static void DeleteDirectoryContentsSafe(string path, bool ignoreErrors =
             DeleteDirectoryContentsSafe(Instance.DirectoryInfo.FromDirectoryName(path), ignoreErrors);
         }
 
+        public static bool IsFileSystemReadOnly(IEnvironment environment)
+        {
+            try
+            {
+                // Temp folder will be clean up everytime site starts up
+                // So it is fine to have leftover folder if deletion failed
+                string folderName = Path.Combine(environment.DataPath, "Temp", Guid.NewGuid().ToString("N"));
+                Instance.Directory.CreateDirectory(folderName);
+                Instance.Directory.Delete(folderName);
+                return false;
```

Comment:
good point! I will wait for Kranthi to confirm, so that can decide the expiration time of the cache. if it require a restart then there is no need for any expiration.